### ISSUE TITLE
Notifiers (and probably Filters as well) should be registered through Dependency Injection

### ISSUE THREAD

**Issue by freerider75** [ElmahCore/ElmahCore#183]
Hi, 
this request is somehow related to my previous #131 : I'm trying to write a notifier that needs to receive and use transient services injected from the DI.

Since the notifiers are created once with new() in their dedicated options properties, they cannot receive constructor parameters provided at runtime by the DI provider, e.g.: the IHttpContextAccessor object, or whatever else.

I think that instead of having a list of "prepared" objects in the ElmahOptions.Notifiers property, all the implementations of IErrorNotifier could be added directly from the services registration and their implementations be received in the elmah middleware constructor as IEnumerable<IErrorNotifier> (assigned to the _notifiers field).

I presume the same change could be done with the filters.

**Comment by Imar** [comment]
Yes, would love this too for the exact reason you mention. I have an IErrorNotifier that takes a custom MailSender which has dependencies like IHttpContextAccessor. With the current implementation, I have to build the services root and fetch my own custom IMailSender and feed it to my notifier in Program.cs. This creates an additional copy of my singleton services which is not great. [freerider75](https://github.com/freerider75)'s PR looks like it's what I need exactly. Any chance this will be accepted for a next release?
